README.md
# node-percipio-contentaccessreport

Retrieve [Content Access Report](https://documentation.skillsoft.com/en_us/percipio/Content/A_Administrator/admn_rpt_content_access.htm) data from Percipio, in JSON format and save locally.

## Requirements

1. A Skillsoft [Percipio](https://www.skillsoft.com/platform-solution/percipio/) Site
1. A [Percipio Service Account](https://documentation.skillsoft.com/en_us/pes/3_services/service_accounts/pes_service_accounts.htm) with permission for accessing [REPORTING API](https://documentation.skillsoft.com/en_us/pes/2_understanding_percipio/rest_api/pes_rest_api.htm)

## Environment Configuration

Once you have copied this repository set the following NODE ENV variables, or config the [.env](.env) file

| ENV       | Required | Description |
| --------- | -------- | ----------- |
| ORGID     | Required | This is the Percipio Organiation UUID for your Percipio Site |
| BEARER    | Required | This is the Percipio Bearer token for a Service Account with permissions for services. |
| BASEURL | Required | This is set to the base URL for the Percipio data center. For US hosted use: https://api.percipio.com For EU hosted use: https://dew1-api.percipio.com |
| TIMEFRAME | Optional | This is a filter criteria that specifies the timeframe for the results.<br/><br/>The report start/end dates are calculated dynamically based on when the report is submitted date.<br/><br/>Options are: DAY, WEEK, THIRTY_DAYS, CALENDAR_MONTH<br/><br/>If left empty/null THIRTY_DAYS is used. |
| START     | Optional | This is a filter criteria that specifies the START date for the report in ISO8601 format.<br/><br/>The END option must be specified if using this.<br/><br/>The TIMEFRAME option must be null if using this.                                                                                     |
| END     | Optional | This is a filter criteria that specifies the END date for the report in ISO8601 format.<br/><br/>The START option must be specified if using this.<br/><br/>The TIMEFRAME option must be null if using this |

## Configuring the API call
Make any additional config changes in [config/default.js](config/default.js) file, to specify the request criteria for the report.

## How to use it

Run the app

```bash
npm start
```

The Percipio [https://api.percipio.com/reporting/api-docs/#/%2Fv1/requestContentActivityReport](https://api.percipio.com/reporting/api-docs/#/%2Fv1/requestContentActivityReport) API wil be called to generate the report.

The Percipio[https://api.percipio.com/reporting/api-docs/#/%2Fv1/getReportRequest](https://api.percipio.com/reporting/api-docs/#/%2Fv1/getReportRequest) API will then be called to download the generated data.

The default configuration returns JSON and the data will be stored in:

```
results/YYYYMMDD_hhmmss_results.json
```

The timestamp component is based on UTC time when the script runs:

| DATEPART | COMMENTS                            |
| -------- | ----------------------------------- |
| YYYY     | Year (i.e. 1970 1971 ... 2029 2030) |
| MM       | Month Number (i.e. 01 02 ... 11 12) |
| DD       | Day (i.e. 01 02 ... 30 31)          |
| HH       | Hour (i.e. 00 01 ... 22 23)         |
| mm       | Minutes (i.e. 00 01 ... 58 59)      |
| ss       | Seconds (i.e. 00 01 ... 58 59)      |

## Changelog

Please see [CHANGELOG](CHANGELOG.md) for more information what has changed recently.

## License

MIT © martinholden-skillsoft


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

Generated by [`auto-changelog`](https://github.com/CookPete/auto-changelog).

## [v1.2.4](https://github.com/martinholden-skillsoft/node-percipio-contentaccessreport/compare/v1.2.3...v1.2.4) - 2021-01-11

### Commits

- Updated Dependencies [`d816af3`](https://github.com/martinholden-skillsoft/node-percipio-contentaccessreport/commit/d816af37c52720a2139ae0e35a9a7b0fe1a2e164)
- ESLINT fixes [`4e6a42d`](https://github.com/martinholden-skillsoft/node-percipio-contentaccessreport/commit/4e6a42d1520dda3a590a3ab57c389d0552123c9c)
- Updated README [`49b83d0`](https://github.com/martinholden-skillsoft/node-percipio-contentaccessreport/commit/49b83d0f7148ccd4a1ee910df9fff6b41f2cf48f)

## [v1.2.3](https://github.com/martinholden-skillsoft/node-percipio-contentaccessreport/compare/v1.2.2...v1.2.3) - 2020-11-04

### Commits

- changelog [`a614a41`](https://github.com/martinholden-skillsoft/node-percipio-contentaccessreport/commit/a614a4111afde2765b7bacd05f2812d96ba59616)
- Updated README [`dd832b2`](https://github.com/martinholden-skillsoft/node-percipio-contentaccessreport/commit/dd832b2d4ad70a1419031153f1849a7954d5c7aa)
- Fix .env config issue [`6bd948b`](https://github.com/martinholden-skillsoft/node-percipio-contentaccessreport/commit/6bd948bcac2bd48867c7bba70d96fcf40c98405d)

## [v1.2.2](https://github.com/martinholden-skillsoft/node-percipio-contentaccessreport/compare/v1.2.1...v1.2.2) - 2020-11-02

### Commits

- Removed extraneous file [`a61322c`](https://github.com/martinholden-skillsoft/node-percipio-contentaccessreport/commit/a61322c49cb75bd08334f682f607552f548b7d88)
- changelog [`615bf72`](https://github.com/martinholden-skillsoft/node-percipio-contentaccessreport/commit/615bf72a9a4ac24c2b3196ecee87a3c5866355bf)

## [v1.2.1](https://github.com/martinholden-skillsoft/node-percipio-contentaccessreport/compare/v1.2.0...v1.2.1) - 2020-11-02

### Commits

- Enable JSON and CSV response handling [`7b04bee`](https://github.com/martinholden-skillsoft/node-percipio-contentaccessreport/commit/7b04bee6480441387fd7f88db28baba5c8e03117)
- changelog [`123b3c7`](https://github.com/martinholden-skillsoft/node-percipio-contentaccessreport/commit/123b3c78110e44a3cc0a4a508daee3d590ecb52e)

## [v1.2.0](https://github.com/martinholden-skillsoft/node-percipio-contentaccessreport/compare/v1.1.0...v1.2.0) - 2020-11-02

### Commits

- Changes to use config and dotenv-safe [`b381b31`](https://github.com/martinholden-skillsoft/node-percipio-contentaccessreport/commit/b381b318f6803d658db302f26c1f75e3c0fcc35e)
- Updated README [`3ba2b17`](https://github.com/martinholden-skillsoft/node-percipio-contentaccessreport/commit/3ba2b179b15eab5133090a2dce8e7d4de21394cd)
- changelog [`e052c54`](https://github.com/martinholden-skillsoft/node-percipio-contentaccessreport/commit/e052c549bd45e7c54570b6299ec5dbabbc776e2f)

## [v1.1.0](https://github.com/martinholden-skillsoft/node-percipio-contentaccessreport/compare/v1.0.0...v1.1.0) - 2020-10-22

### Commits

- changelog [`51ee68f`](https://github.com/martinholden-skillsoft/node-percipio-contentaccessreport/commit/51ee68fda0e1a15a8436b3623770b07e9d4ed64a)
- Fixed Bad Polling Retry Options [`0107d5c`](https://github.com/martinholden-skillsoft/node-percipio-contentaccessreport/commit/0107d5ccef4d151275b7a195f9477978648af730)
- Fixed Typo [`a6a27b7`](https://github.com/martinholden-skillsoft/node-percipio-contentaccessreport/commit/a6a27b71599373d0d8783cab1075a8ad7a95566c)

## v1.0.0 - 2020-10-22

### Commits

- Initial commit [`06125db`](https://github.com/martinholden-skillsoft/node-percipio-contentaccessreport/commit/06125db7f098f78925ccca03f27d6b3a807a4146)


config/config.global.js
const moment = require('moment');

const config = {};

// Indicates a name for the configuration
config.customer = 'none';
config.startTimestamp = moment().utc().format('YYYYMMDD_HHmmss');

// DEBUG Options - Enables the check for Fiddler, if running the traffic is routed thru Fiddler
config.debug = {};
// Debug logging
// One of the supported default logging levels for winston - see https://github.com/winstonjs/winston#logging-levels
config.debug.loggingLevel = 'info';
config.debug.path = 'logs';
config.debug.filename = `app_${config.startTimestamp}.log`;

config.output = {};
// Output path
config.output.path = 'results/output';
// Filename
config.output.filename = `${config.startTimestamp}_results.json`;

// Report Generation Request
config.reportrequest = {};
// Request Timeout
config.reportrequest.timeout = 2000;
// Bearer Token
config.reportrequest.bearer = null;
// Base URI to Percipio API
config.reportrequest.baseuri = process.env.EUDC
  ? 'https://dew1-api.percipio.com'
  : 'https://api.percipio.com';
// Request Path Parameters
config.reportrequest.path = {};
/**
 * Name: orgId
 * Description: Organization UUID
 * Required: true
 * Type: string
 * Format: uuid
 */
config.reportrequest.path.orgId = null;
// Request Query string Parameters
config.reportrequest.query = {};
// Request Body
config.reportrequest.body = null;
// Method
config.reportrequest.method = 'post';
// The Service Path
config.reportrequest.uri = `${config.reportrequest.baseuri}/reporting/v1/organizations/${config.reportrequest.path.orgId}/report-requests/content-access`;

// Request
config.pollrequest = {};
// Request Timeout
config.pollrequest.timeout = 2000;
// Bearer Token
config.pollrequest.bearer = null;
// Base URI to Percipio API
config.pollrequest.baseuri = process.env.EUDC
  ? 'https://dew1-api.percipio.com'
  : 'https://api.percipio.com';
// Request Path Parameters
config.pollrequest.path = {};
/**
 * Name: orgId
 * Description: Organization UUID
 * Required: true
 * Type: string
 * Format: uuid
 */
config.pollrequest.path.orgId = null;
// Request Query string Parameters
config.pollrequest.query = {};
// Request Body
config.pollrequest.body = null;
// Method
config.pollrequest.method = 'get';
// The Service Path
config.pollrequest.uri = `${config.pollrequest.baseuri}/reporting/v1/organizations/${config.pollrequest.path.orgId}/report-requests/${config.pollrequest.path.reportRequestId}`;

// Global Web Retry Options for promise retry
// see https://github.com/IndigoUnited/node-promise-retry#readme
config.retry_options = {};
config.retry_options.retries = 3;
config.retry_options.minTimeout = 1000;
config.retry_options.maxTimeout = 2000;

/*
Polling options for retrying report availability
see https://github.com/IndigoUnited/node-promise-retry#readme
options is a JS object that can contain any of the following keys:
retries: The maximum amount of times to retry the operation.Default is 10.
Seting this to 1 means do it once, then retry it once.
factor: The exponential factor to use.Default is 2.
minTimeout: The number of milliseconds before starting the first retry.Default is 1000.
maxTimeout: The maximum number of milliseconds between two retries.Default is Infinity.
randomize: Randomizes the timeouts by multiplying with a factor between 1 to 2. Default is false.
*/

config.polling_options = {};
config.polling_options.retries = 20;
config.polling_options.minTimeout = 60 * 1000;
config.polling_options.maxTimeout = Infinity;
// Using Factor=1 means polling at fixed interval of minTimeout
config.polling_options.factor = 1;

module.exports = config;


config/config.default.js
const config = require('./config.global');

config.customer = 'default';

// Debug logging
// One of the supported default logging levels for winston - see https://github.com/winstonjs/winston#logging-levels
// config.debug.loggingLevel = 'debug';
config.debug.path = 'results/output';
config.debug.filename = `${config.customer}_${config.startTimestamp}.log`;

// Output path
config.output.path = 'results/output';
config.output.filename = `${config.customer}_contentaccess_${config.startTimestamp}.json`;

// Report Generation Request
// Request
config.reportrequest = {};
// Timeout
config.reportrequest.timeout = 2000;

// Bearer Token
config.reportrequest.bearer = process.env.BEARER || null;
// Base URI to Percipio API
config.reportrequest.baseuri = process.env.EUDC
  ? 'https://dew1-api.percipio.com'
  : 'https://api.percipio.com';
// Request Path Parameters
config.reportrequest.path = {};
/**
 * Name: orgId
 * Description : Organization UUID
 * Required: true
 * Type: string
 * Format: uuid
 */
config.reportrequest.path.orgId = process.env.ORGID || null;

// Request Query string Parameters
config.reportrequest.query = {};

// Request Body
config.reportrequest.body = {};
/**
 * Name: start
 * Description : Start date for events retrieval
 * Type: string
 * Format: date-time
 * Example: 2018-01-01T10:10:24Z
 */
config.reportrequest.body.start = process.env.START || null;
/**
 * Name: end
 * Description : End date for events retrieval
 * Type: string
 * Format: date-time
 * Example: 2018-01-10T10:20:24Z
 */
config.reportrequest.body.end = process.env.END || null;
/**
 * Name: timeFrame
 * Description : To calculate the start/end date dynamically based on timeframe and when the
 * report is submitted date. [User can submit an absolute date range (start/end date) or a
 * relative date range (timeframe) but never both]
 * Type: string
 * Enum: DAY, WEEK, THIRTY_DAYS, CALENDAR_MONTH
 */
config.reportrequest.body.timeFrame = config.reportrequest.body.start
  ? null
  : process.env.TIMEFRAME || 'THIRTY_DAYS';
/**
 * Name: audience
 * Description : Audience filter, defaults to all audience
 * Type: string
 */
config.reportrequest.body.audience = 'ALL';
/**
 * Name: locale
 * Description : The field to use for specifying language. Will default to all when not
 * specified. example format 'en' or 'fr' or 'de' etc.
 * Type: string
 */
config.reportrequest.body.locale = null;
/**
 * Name: collectionName
 * Description : Collections used for content lookup, comma delimited and defaults to all
 * collections
 * Type: string
 */
config.reportrequest.body.collectionName = 'ALL';
/**
 * Name: assetTypes
 * Description : Content types filter, comma delimited and default to all content types
 * Type: string
 * Example: AudioBook,Audio Summary,Book,Book Summary,Course,Linked Content,Video
 */
config.reportrequest.body.assetTypes = null;
/**
 * Name: sort
 * Description : The field to use for sorting and which order to sort in, if sort is not
 * included the results will be returned descending by total accesses
 * Type: object
 */
config.reportrequest.body.sort = {};

/**
 * Name: field
 * Type: string
 * Enum: collection, contentTitle, channelTitle, contentId, contentType, uniqueUserAccesses,
 * totalAccesses, totalDuration, timesAccessed, avgDuration, totalCompletions
 */
config.reportrequest.body.sort.field = 'totalAccesses';
/**
 * Name: order
 * Type: string
 * Enum: asc, desc
 */
config.reportrequest.body.sort.order = 'desc';
/**
 * Name: status
 * Description : Content access status filter, defaults to all status type
 * Type: string
 * Enum: Published, Retired, PendingRetirement
 */
config.reportrequest.body.status = null;
/**
 * Name: sftpId
 * Description : SFTP Id associated with OrgId
 * Type: string
 * Format: uuid
 */
config.reportrequest.body.sftpId = null;
/**
 * Name: isFileRequiredInSftp
 * Description : Generated files are required to deliver in the respected sftp location.
 * Default value is true.
 * Type: boolean
 */
config.reportrequest.body.isFileRequiredInSftp = false;
/**
 * Name: zip
 * Description : Generate the reports in zip file format. Default value is false.
 * Type: boolean
 */
config.reportrequest.body.zip = null;
/**
 * Name: encrypt
 * Description : Generate the report file as PGP encrypted file. Default value is false.
 * Type: boolean
 */
config.reportrequest.body.encrypt = null;
/**
 * Name: formatType
 * Description : Format Type, defaults to JSON, the value is extracted from Accept attribute
 * in header
 * Type: string
 * Enum: JSON, CSV, TXT
 */
config.reportrequest.body.formatType = 'JSON';
/**
 * Name: fileMask
 * Description : Absolute or masked pattern for the generated report file. Example file masks
 * - fileName_{DD}{MM}{YYYY}, fileName_{ORG_ID}
 * Type: string
 */
config.reportrequest.body.fileMask = null;
/**
 * Name: folderName
 * Description : custom folder under sftp reports wherein the generated report file is to be
 * placed.
 * Type: string
 */
config.reportrequest.body.folderName = null;
/**
 * Name: includeMillisInFilename
 * Description : Generate files with unix based timestamp. Example -
 * fileName.csv.1561642446608
 * Type: boolean
 */
config.reportrequest.body.includeMillisInFilename = null;

// Method
config.reportrequest.method = 'post';
// The Service Path
config.reportrequest.uri = `${config.reportrequest.baseuri}/reporting/v1/organizations/${config.reportrequest.path.orgId}/report-requests/content-access`;

// Report Polling Request
config.pollrequest = {};
// Request Timeout
config.pollrequest.timeout = 2000;
// Bearer Token
config.pollrequest.bearer = process.env.BEARER || null;
// Base URI to Percipio API
config.pollrequest.baseuri = process.env.EUDC
  ? 'https://dew1-api.percipio.com'
  : 'https://api.percipio.com';
// Request Path Parameters
config.pollrequest.path = {};
/**
 * Name: orgId
 * Description : Organization UUID
 * Required: true
 * Type: string
 * Format: uuid
 */
config.pollrequest.path.orgId = process.env.ORGID || null;
/**
 * Name: reportRequestId
 * Description : Handle to access the report content
 * Required: true
 * Type: string
 */
config.pollrequest.path.reportRequestId = null;

// Request Query string Parameters
config.pollrequest.query = {};

// Request Body
config.pollrequest.body = null;

// Method
config.pollrequest.method = 'get';
// The Service Path
config.pollrequest.uri = `${config.pollrequest.baseuri}/reporting/v1/organizations/${config.pollrequest.path.orgId}/report-requests/${config.pollrequest.path.reportRequestId}`;

module.exports = config;


config/default.js
const moment = require('moment');
const defer = require('config/defer').deferConfig;

const config = {};

// Indicates a name for the configuration
config.customer = 'default';
config.startTimestamp = moment().utc().format('YYYYMMDD_HHmmss');
// Null logger object
config.logger = {};

// DEBUG Options - Enables the check for Fiddler, if running the traffic is routed thru Fiddler
config.debug = {};
// Debug logging
// One of the supported default logging levels for winston - see https://github.com/winstonjs/winston#logging-levels
config.debug.loggingLevel = 'info';
config.debug.path = 'results';
config.debug.filename = defer((cfg) => `${cfg.startTimestamp}_results.log`);

config.output = {};
// Output path
config.output.path = 'results';
// Filename
config.output.filename = defer((cfg) => `${cfg.startTimestamp}_results.json`);

// Report Generation Request
config.reportrequest = {};
// Request Timeout
config.reportrequest.timeout = 2000;
// Bearer Token
config.reportrequest.bearer = null;
// Base URI to Percipio API
config.reportrequest.baseURL = null;
// Request Path Parameters
config.reportrequest.path = {};
/**
 * Name: orgId
 * Description: Organization UUID
 * Required: true
 * Type: string
 * Format: uuid
 */
config.reportrequest.path.orgId = null;

// Request Query string Parameters
config.reportrequest.query = {};

// Request Body
config.reportrequest.body = {};
/**
 * Name: start
 * Description : Start date for events retrieval
 * Type: string
 * Format: date-time
 * Example: 2018-01-01T10:10:24Z
 */
config.reportrequest.body.start = process.env.START || null;
/**
 * Name: end
 * Description : End date for events retrieval
 * Type: string
 * Format: date-time
 * Example: 2018-01-10T10:20:24Z
 */
config.reportrequest.body.end = process.env.END || null;
/**
 * Name: timeFrame
 * Description : To calculate the start/end date dynamically based on timeframe and when the
 * report is submitted date. [User can submit an absolute date range (start/end date) or a
 * relative date range (timeframe) but never both]
 * Type: string
 * Enum: DAY, WEEK, THIRTY_DAYS, CALENDAR_MONTH
 */
config.reportrequest.body.timeFrame = config.reportrequest.body.start
  ? null
  : process.env.TIMEFRAME || 'THIRTY_DAYS';
/**
 * Name: audience
 * Description : Audience filter, defaults to all audience
 * Type: string
 */
config.reportrequest.body.audience = 'ALL';
/**
 * Name: locale
 * Description : The field to use for specifying language. Will default to all when not
 * specified. example format 'en' or 'fr' or 'de' etc.
 * Type: string
 */
config.reportrequest.body.locale = null;
/**
 * Name: collectionName
 * Description : Collections used for content lookup, comma delimited and defaults to all
 * collections
 * Type: string
 */
config.reportrequest.body.collectionName = 'ALL';
/**
 * Name: assetTypes
 * Description : Content types filter, comma delimited and default to all content types
 * Type: string
 * Example: AudioBook,Audio Summary,Book,Book Summary,Course,Linked Content,Video
 */
config.reportrequest.body.assetTypes = null;
/**
 * Name: sort
 * Description : The field to use for sorting and which order to sort in, if sort is not
 * included the results will be returned descending by total accesses
 * Type: object
 */
config.reportrequest.body.sort = {};

/**
 * Name: field
 * Type: string
 * Enum: collection, contentTitle, channelTitle, contentId, contentType, uniqueUserAccesses,
 * totalAccesses, totalDuration, timesAccessed, avgDuration, totalCompletions
 */
config.reportrequest.body.sort.field = 'totalAccesses';
/**
 * Name: order
 * Type: string
 * Enum: asc, desc
 */
config.reportrequest.body.sort.order = 'desc';
/**
 * Name: status
 * Description : Content access status filter, defaults to all status type
 * Type: string
 * Enum: Published, Retired, PendingRetirement
 */
config.reportrequest.body.status = null;
/**
 * Name: sftpId
 * Description : SFTP Id associated with OrgId
 * Type: string
 * Format: uuid
 */
config.reportrequest.body.sftpId = null;
/**
 * Name: isFileRequiredInSftp
 * Description : Generated files are required to deliver in the respected sftp location.
 * Default value is true.
 * Type: boolean
 */
config.reportrequest.body.isFileRequiredInSftp = false;
/**
 * Name: zip
 * Description : Generate the reports in zip file format. Default value is false.
 * Type: boolean
 */
config.reportrequest.body.zip = null;
/**
 * Name: encrypt
 * Description : Generate the report file as PGP encrypted file. Default value is false.
 * Type: boolean
 */
config.reportrequest.body.encrypt = null;
/**
 * Name: formatType
 * Description : Format Type, defaults to JSON, the value is extracted from Accept attribute
 * in header
 * Type: string
 * Enum: JSON, CSV, TXT
 */
config.reportrequest.body.formatType = 'JSON';
/**
 * Name: fileMask
 * Description : Absolute or masked pattern for the generated report file. Example file masks
 * - fileName_{DD}{MM}{YYYY}, fileName_{ORG_ID}
 * Type: string
 */
config.reportrequest.body.fileMask = null;
/**
 * Name: folderName
 * Description : custom folder under sftp reports wherein the generated report file is to be
 * placed.
 * Type: string
 */
config.reportrequest.body.folderName = null;
/**
 * Name: includeMillisInFilename
 * Description : Generate files with unix based timestamp. Example -
 * fileName.csv.1561642446608
 * Type: boolean
 */
config.reportrequest.body.includeMillisInFilename = null;

// Method
config.reportrequest.method = 'post';
// The Service Path the placeholders {} are replaced by values from *.path
config.reportrequest.uritemplate =
  '/reporting/v1/organizations/{orgId}/report-requests/content-access';

// Request
config.pollrequest = {};
// Request Timeout
config.pollrequest.timeout = 2000;
// Bearer Token
config.pollrequest.bearer = null;
// Base URI to Percipio API
config.pollrequest.baseURL = null;
// Request Path Parameters
config.pollrequest.path = {};
/**
 * Name: orgId
 * Description: Organization UUID
 * Required: true
 * Type: string
 * Format: uuid
 */
config.pollrequest.path.orgId = null;
/**
 * Name: reportRequestId
 * Description : Handle to access the report content
 * Required: true
 * Type: string
 */
config.pollrequest.path.reportRequestId = null;
// Request Query string Parameters
config.pollrequest.query = {};
// Request Body
config.pollrequest.body = null;
// Method
config.pollrequest.method = 'get';
// The Service Path the placeholders {} are replaced by values from *.path
config.pollrequest.uritemplate =
  '/reporting/v1/organizations/{orgId}/report-requests/{reportRequestId}';

// Global Web Retry Options for promise retry
// see https://github.com/IndigoUnited/node-promise-retry#readme
config.retry_options = {};
config.retry_options.retries = 3;
config.retry_options.minTimeout = 1000;
config.retry_options.maxTimeout = 2000;

/*
Polling options for retrying report availability
see https://github.com/IndigoUnited/node-promise-retry#readme
options is a JS object that can contain any of the following keys:
retries: The maximum amount of times to retry the operation.Default is 10.
Seting this to 1 means do it once, then retry it once.
factor: The exponential factor to use.Default is 2.
minTimeout: The number of milliseconds before starting the first retry.Default is 1000.
maxTimeout: The maximum number of milliseconds between two retries.Default is Infinity.
randomize: Randomizes the timeouts by multiplying with a factor between 1 to 2. Default is false.
*/

config.polling_options = {};
config.polling_options.retries = 20;
config.polling_options.minTimeout = 60 * 1000;
config.polling_options.maxTimeout = Infinity;
// Using Factor=1 means polling at fixed interval of minTimeout
config.polling_options.factor = 1;

module.exports = config;
